plugin/kvictoria/kvictoria.go

```diff
@@ -0,0 +1,243 @@
+}
+
+// OnGroupManageError implements the [kgo.HookBrokerThrottle] interface for metrics gathering.
+// This method is meant to be called by the hook system and not by the user.
+func (m *Metrics) OnGroupManageError(err error) {
+	labels := make(map[string]string)
+
+	var kerr *kerr.Error
+	if errors.As(err, &kerr) {
+		labels["error_message"] = kerr.Message
+	}
+
+	vm.GetOrCreateCounter(m.buildName("group_manage_error", labels)).Inc()
+}
+
+// OnProduceBatchWritten implements the [kgo.HookProduceBatchWritten] interface for metrics gathering.
+// This method is meant to be called by the hook system and not by the user.
+func (m *Metrics) OnProduceBatchWritten(meta kgo.BrokerMetadata, topic string, partition int32, metrics kgo.ProduceBatchMetrics) {
+	labels := map[string]string{
+		"node_id":   kgo.NodeName(meta.NodeID),
+		"topic":     topic,
+		"partition": strconv.FormatInt(int64(partition), 10),
+	}
+
+	vm.GetOrCreateCounter(m.buildName("produce_uncompressed_bytes_total", labels)).Add(metrics.UncompressedBytes)
+	vm.GetOrCreateCounter(m.buildName("produce_compressed_bytes_total", labels)).Add(metrics.CompressedBytes)
+	vm.GetOrCreateCounter(m.buildName("produce_batches_total", labels)).Inc()
+	vm.GetOrCreateCounter(m.buildName("produce_records_total", labels)).Add(metrics.NumRecords)
+}
+
+// OnFetchBatchRead implements the [kgo.HookFetchBatchRead] interface for metrics gathering.
+// This method is meant to be called by the hook system and not by the user.
+func (m *Metrics) OnFetchBatchRead(meta kgo.BrokerMetadata, topic string, partition int32, metrics kgo.FetchBatchMetrics) {
+	labels := map[string]string{
+		"node_id":   kgo.NodeName(meta.NodeID),
+		"topic":     topic,
+		"partition": strconv.FormatInt(int64(partition), 10),
+	}
+
+	vm.GetOrCreateCounter(m.buildName("fetch_uncompressed_bytes_total", labels)).Add(metrics.UncompressedBytes)
+	vm.GetOrCreateCounter(m.buildName("fetch_compressed_bytes_total", labels)).Add(metrics.CompressedBytes)
+	vm.GetOrCreateCounter(m.buildName("fetch_batches_total", labels)).Inc()
+	vm.GetOrCreateCounter(m.buildName("fetch_records_total", labels)).Add(metrics.NumRecords)
+}
+
+// buildName constructs a metric name for the VictoriaMetrics metrics library.
+//
+// The library expects the user to create a metric for each and every variation of a metric
+// by providing the full name, including labels: there is no equivalent to the *Vec variants
+// in the official Prometheus client.
+//
+// This function is a helper to build such a name, taking care of properly adding
+// the namespace if present, subsystem if present and labels if present.
+func (m *Metrics) buildName(name string, labels map[string]string) string {
+	var builder strings.Builder
+
+	if m.cfg.namespace != "" {
+		builder.WriteString(m.cfg.namespace + "_")
+	}
+	if m.cfg.subsystem != "" {
+		builder.WriteString(m.cfg.subsystem + "_")
+	}
+	builder.WriteString(name)
+
+	// Note: can't use maps.Keys yet because it needs Go 1.23+
```

Comment:
You can enforce this in the go.mod to allow the import (I usually leave kgo with the oldest Go support, but other packages or plugins I allow being higher).

Reply:
good idea